test/http_headers_test.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io' as io;
+
+import 'package:cronet/cronet.dart';
+import 'package:test/test.dart';
+
+const host = 'localhost';
+const sentData = 'Hello, world!';
+
+void main() {
+  group('Header Tests', () {
+    late HttpClient client;
+    late io.HttpServer server;
+    late int port;
+    setUp(() async {
+      client = HttpClient();
+      server = await io.HttpServer.bind(io.InternetAddress.anyIPv6, 0);
+      port = server.port;
+      server.listen((io.HttpRequest request) {
+        request.response.write(request.headers.value('test-header'));
+        request.response.close();
+      });
+    });
+
+    test('Send an arbitrary http header to the server', () async {
+      final request = await client.getUrl(Uri.parse('http://$host:$port/'));
+      request.headers.set('test-header', sentData);
```

Comment:
Maybe we should also provide users a way to set multiple headers? What do you think?
I think a very popular use case would be to have a standard set of headers which is used in multiple requests to an API.

Reply:
That will be handy indeed :).

We can plan to introduce 2 new apis in a later pr:

* `headers.setAll` that will take a map of header name and value.
* An API to set headers to `HttpClient`  that all the initiated `HttpClientRequest` will use by default.

These 2 APIs aren't part of `dart:io` as of now. But, will be a really nice addition to this package.

Easy way of adding few common headers will be added via #27 to comply with `dart:io` apis.

Reply:
Cool. 😀